README.md

```diff
@@ -42,6 +42,104 @@ This will construct the platform specific build environment. For example, on
 Windows with a properly configured Visual Studio, this will create in the
 directory `build.d` the solution file `Resonance.sln`.
 
+# Example
+
+The following example is adapted from the `RunMatrixMultiply` test of the ONNX
+model support. This is one of two built-in implementations of
+`inference::Model`. ONNX models contain within themselves a description of
+their input and output schemas, including their types. This example is
+structured more for simplicity of illustrating the key structures, rather
+complete coverage.
+
+First, we start with a pre-amble where we simply load the model.
+
+```cpp
+auto path = test_dir_path + "matmul.onnx";
+auto modelExpected = onnx_model::Model::Load(path);
+auto model = modelExpected.value();
+```
+
+Note that in this first phase we are loading a model. This code is peculiar to
+the `onnx_model` namespace, but we already see some common conventions emerge.
+In particular, throughout this codebase there is a lot of usage of things like
+`tl::expected<T, std::error_code>` for code that *might* fail. (In this case,
+`T` would be a `std::shared_ptr<Model>`.) One can read more about this
+[structure here](https://github.com/TartanLlama/expected).
+
+Throughout this code you'll note that we are living very dangerously, by not
+actually checking to see whether the `tl::expected` actually has the values we
+expect. In real production code a user would want to check and provide the
+remediation that is appropriate for their library (whether that is raising an
+exception, returning an error code, or something else is up to the user).
+
+The particular model that is being loaded is an ONNX model that takes two
+placeholders (in the ONNX model sense) named `A` and `B`, and produces another
+output `C`, that is the result of performing a matrix multiplication of those
+two inputs.
+
+```cpp
+FeatureBroker fb(model);
+// Binding inputs and binding outputs.
+auto inputA = fb.BindInput<Tensor<double>>("A:0").value_or(nullptr);
+auto inputB = fb.BindInput<Tensor<double>>("B:0").value_or(nullptr);
+auto output = fb.BindOutput<Tensor<double>>("C:0").value_or(nullptr);
+```
+
+In this particular section we create the `FeatureBroker` *and* associate it
+with a model in the first step. The inputs and outputs are bound, fed, and
+consumed in all the same function so it is simplest to do things that way, but
+in other scenarios it is often common to create a feature broker without any
+model at all, bind inputs to that, and then later on the actual inferencing

```

Comment:
Suggestion: It may be worth mentioning the practical scenario that you had in mind. I'm assuming you meant the scenario where the feature is produced in a different component than the one where the inferencing will take place? Perhaps a simple example might make this statement less abstract.

Reply:
As a first example, especially one appearing in the `README.md`, I'm not sure that it is a good idea to go into a more complex example. On the other hand, given that the library was built to support that scenario, I can definitely see the point in adding it, though I might prefer it was "chapter 2" of an example rather than being the first example ever.

Reply:
So perhaps we'll leave the simpler example as is, and if more examples are added later, that should be fine?

Reply:
I did make the paragraphs a bit more separated so it's clear I'm talking about two separate things, since the sentence was pretty long...

Reply:
Yeah, agree. This makes sense and the rephrase reads well.